### PR TITLE
SetupAuthoring for Roslyn localization

### PR DESCRIPTION
**Pull request by Cosifne**
Related [issue](https://devdiv.visualstudio.com/DevDiv/_workitems/edit/723558)

Generally I just follow the pattern of Tom's previous commits.
And add resources file for each Vsix appears in the SignToolData.json

Plus Tom's changes, this PR currently includes

- [Diagnostics Window]
- [Roslyn compiler extension]
- [Expression Elevator]
- [Interactive Components ] 
- [Language Services]
- [Compilers]
 
Updates:
Talked to Tom and I will try to understand the build process of Compilers packages and wait for Tomas comes back to get fully understand the build procedure

### CODE REVIEW

File: src/Setup/DevDivVsix/RoslynCompilersExtension/Roslyn.Compilers.Extension.Resources.AllLangsNoEnu.swr [hunk 1]
```diff
@@ -0,0 +1,7 @@
+folder "TBD"
```
Comment (Cosifne) [inline src/Setup/DevDivVsix/RoslynCompilersExtension/Roslyn.Compilers.Extension.Resources.AllLangsNoEnu.swr]: @tmeschter 
Not sure where it should be installed.

File: src/Setup/DevDivVsix/RoslynVisualStudioDiagnosticsWindow/Roslyn.VisualStudio.Diagnostics.Window.Resources.AllLangsNoEnu.swr [hunk 2]
```diff
@@ -0,0 +1,2 @@
+folder "TBD"
```
Comment (Cosifne) [inline src/Setup/DevDivVsix/RoslynVisualStudioDiagnosticsWindow/Roslyn.VisualStudio.Diagnostics.Window.Resources.AllLangsNoEnu.swr]: @tmeschter 
Not sure where it should be installed

File: src/Setup/DevDivVsix/RoslynCompilersExtension/Roslyn.Compilers.Extension.Resources.AllLangsNoEnu.swr [hunk 3]
```diff
@@ -0,0 +1,7 @@
...
+  file source=$(OutputPath)\Dlls\Microsoft.Build.Tasks.CodeAnalysis\netcoreapp2.0\$(Culture)\Microsoft.Build.Tasks.CodeAnalysis.resources.dll
```
Comment (Cosifne) [inline src/Setup/DevDivVsix/RoslynCompilersExtension/Roslyn.Compilers.Extension.Resources.AllLangsNoEnu.swr]: This resources file exists both under netcoreapp2.0 and net46 folder. Not sure which one should get included

File: src/Setup/DevDivVsix/RoslynVisualStudioDiagnosticsWindow/Roslyn.VisualStudio.Diagnostics.Window.Resources.AllLangsNoEnu.swr [hunk 4]
```diff
@@ -0,0 +1,2 @@
...
+  file source=$(OutputPath)\Vsix\Roslyn.VisualStudio.DiagnosticsWindow\$(Culture)\Microsoft.Build.Tasks.CodeAnalysis.resources.dll
```
Comment (Cosifne) [inline src/Setup/DevDivVsix/RoslynVisualStudioDiagnosticsWindow/Roslyn.VisualStudio.Diagnostics.Window.Resources.AllLangsNoEnu.swr]: @tmeschter 
I can't find this assembly under the output/Dlls or output/Exes, can I just reference it under Vsix folder?
Also I find that a lot of resources assemblies are seems not generated by the compilation of Rolsyn, should I also included them?

**Comment by tmeschter** [comment]
We don't need to ship satellite assemblies for things that don't ship with VS. Specifically, I don't think we need them for the Diagnostics Window (I don't even know what this is) or the Roslyn compiler extension (that's a "nice to have" but we shouldn't deal with it now).

**Comment by tmeschter** [comment]
@Cosifne A lot of the dotnet/roslyn infrastructure has changed and you won't be able to re-use my existing commits in this way. Rather, they should just serve as a guide to what needs to be done.

File: build/scripts/build.ps1 [hunk 5]
```diff
@@ -300,6 +300,38 @@ function Build-ExtraSignArtifacts() {
...
+
+        Write-Host "Building Roslyn.Compilers.Extension Resources Swixes"
+        foreach ($language in $vsLanguages) {
+            Run-MSBuild "DevDivVsix\RoslynCompilersExtension\Roslyn.Compilers.Extension.Resources.swixproj" "/p:Lang=$($language)"
+        }
+
+        Write-Host "Building Roslyn.Compilers.Extension Resources Swixes"
+        foreach ($language in $vsLanguages) {
+            Run-MSBuild "DevDivVsix\RoslynVisualStudioDiagnosticsWindow\Roslyn.VisualStudio.Diagnostics.Window.Resources.swixproj" "/p:Lang=$($language)"
+        }
```
Comment (jaredpar) [inline build/scripts/build.ps1]: How much time does this add to the builds? Seems fairly significant. Is it possible for us to reduce the number of MSBuild invocations here? Possibly by creating an uber project of some sort.

File: src/Setup/DevDivVsix/CompilersPackage/Microsoft.CodeAnalysis.Compilers.Resources.AllLangsNoEnu.swr [hunk 6]
```diff
@@ -0,0 +1,8 @@
+folder "InstallDir:\MSBuild\15.0\Bin\Roslyn\$(Culture)"
+  file source=$(OutputPath)\Dlls\Microsoft.CodeAnalysis\$(Culture)\Microsoft.CodeAnalysis.resources.dll
+  file source=$(OutputPath)\Dlls\Microsoft.CodeAnalysis.CSharp\$(Culture)\Microsoft.CodeAnalysis.CSharp.resources.dll
+  file source=$(OutputPath)\Dlls\Microsoft.CodeAnalysis.VisualBasic\$(Culture)\Microsoft.CodeAnalysis.VisualBasic.resources.dll
+  file source=$(OutputPath)\Dlls\Microsoft.CodeAnalysis.Scripting\$(Culture)\Microsoft.CodeAnalysis.Scripting.resources.dll
+  file source=$(OutputPath)\Dlls\Microsoft.CodeAnalysis.CSharp.Scripting\$(Culture)\Microsoft.CodeAnalysis.CSharp.Scripting.resources.dll
+  file source=$(OutputPath)\Dlls\Microsoft.Build.Tasks.CodeAnalysis\net46\$(Culture)\Microsoft.Build.Tasks.CodeAnalysis.resources.dll
```
Comment (jaredpar) [inline src/Setup/DevDivVsix/CompilersPackage/Microsoft.CodeAnalysis.Compilers.Resources.AllLangsNoEnu.swr]: What validation do we get when these entries are wrong? For instance the net46 folder will change very soon due to Dev16 changes. Will the build break when there is no longer output to the net46 folder (hoping the answer is yes)?

File: src/Setup/DevDivVsix/MicrosoftCodeAnalysisExpressionEvaluator/Microsoft.CodeAnalysis.ExpressionEvaluator.Resources.swr [hunk 7]
```diff
@@ -0,0 +1,7 @@
+use vs
+
+package name=Microsoft.CodeAnalysis.ExpressionEvaluator.Resources
+        version=$(Version)
+        vs.package.language=$(LangCountry)
+
+# No files to work around bug #239676
```
Comment (jaredpar) [inline src/Setup/DevDivVsix/MicrosoftCodeAnalysisExpressionEvaluator/Microsoft.CodeAnalysis.ExpressionEvaluator.Resources.swr]: Please include a URL to the issue. Issue numbers by themselves aren't sufficient for tracking down bugs.

File: src/Setup/DevDivVsix/RoslynCompilersExtension/Roslyn.Compilers.Extension.Resources.swr [hunk 8]
```diff
@@ -0,0 +1,8 @@
+use vs
+
+package name=Roslyn.Compilers.Extension.Resources
+        version=$(Version)
+        vs.package.language=$(LangCountry)
+
+# No files to work around bug #239676
```
Comment (jaredpar) [inline src/Setup/DevDivVsix/RoslynCompilersExtension/Roslyn.Compilers.Extension.Resources.swr]: Same about URL

**Pull request closed without merging**